From 8084885e55bdc7e9f05bc25b76b8776a0205193d Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@ycbcr.xyz>
Date: Mon, 25 May 2020 16:27:51 +0200
Subject: [PATCH 2/3] configure: don't error on GetThreadErrorMode if we're not
 going to use it

In UWP builds GetThreadErrorMode is not supported (at least not on all Win8/Win10).
But it's still possible to build mpg123 without module handling.
---
 configure.ac | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/configure.ac b/configure.ac
index 917d251..c494221 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2435,6 +2435,7 @@ if test x$win32_specific_codes = xenabled; then
     AC_MSG_RESULT([no])
   fi
 # Check GetThreadErrorMode
+  if test x"$modules" != xdisabled; then
   AC_MSG_CHECKING([if we have GetThreadErrorMode])
     AC_LINK_IFELSE([AC_LANG_SOURCE([
     #include <windows.h>
@@ -2457,6 +2458,7 @@ if test x$win32_specific_codes = xenabled; then
     AC_MSG_RESULT([no])
     AC_ERROR([GetThreadErrorMode is required but not found])
   fi
+  fi
 fi
 
 #### WINVER Bump
-- 
2.26.0.windows.1

